### PR TITLE
Add more linting

### PR DESCRIPTION
**Pull request by tac0turtle**
- added more linting

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

### CODE REVIEW

File: binary-decode.go [hunk 1]
```diff
@@ -30,7 +30,9 @@ const (
 // CONTRACT: rv.CanAddr() is true.
 func (cdc *Codec) decodeReflectBinary(bz []byte, info *TypeInfo,
 	rv reflect.Value, fopts FieldOptions, bare bool) (n int, err error) {
+
 	if !rv.CanAddr() {
+		fmt.Println(bz, "bz")
```
Comment (melekes) [inline binary-decode.go]: ???

**Review by melekes — APPROVED** [review]
👍

**Review by tessr — APPROVED** [review]

File: amino.go [hunk 2]
```diff
@@ -331,7 +331,7 @@ func (cdc *Codec) UnmarshalBinaryLengthPrefixedReader(r io.Reader, ptr interface
 	}
 	l = int64(u64)
 	if l < 0 {
-		_ = errors.Errorf(
+		_ = errors.Errorf( //nolint:errcheck
```
Comment (melekes) [inline amino.go:334]: what is this?)

Comment (tac0turtle) [reply amino.go:334]: I was looking at this as well, with amino being a bit touchy I didn't want to change to much stuff in this PR so I left it

Comment (melekes) [reply amino.go:334]: Right. But this code does not do anything, so maybe we should comment it completely.

**Pull request merged**